packages/storage/amplify_storage_s3_dart/lib/src/exception/s3_storage_exception.dart

```diff
@@ -38,8 +40,13 @@ class S3StorageException extends StorageException {
       case 403:
         return S3StorageException(
           'S3 access denied when making the API call.',
-          recoverySuggestion:
-              'Please ensure that correct `StorageAccessLevel` and `targetIdentityId` are included in the options.',
+          recoverySuggestion: _clientErrorRecoveryMessage,
+          underlyingException: exception,
+        );
+      case 404:
+        return S3StorageException(
+          'Object is not found in the bucket.',
+          recoverySuggestion: _clientErrorRecoveryMessage,
```

Comment:
Is this the correct recovery suggestion?

Reply:
I'll expand `_clientErrorRecoveryMessage` to include "correct object key" as well.